### PR TITLE
fix: enhance boilerplate infra & update docs

### PR DESCRIPTION
**Pull request by agdimech**
- Enhance boilerplate infrastructure in each language to support calling the API for authenticated Cognito Roles.
- Updated docs

### CODE REVIEW

File: docs/content/getting_started/index.md [hunk 1]
```diff
@@ -101,7 +101,49 @@ The AWS PDK lets you define your project structure as code in one of its support
 
 The AWS PDK has first-class support for Typescript, Python, & Java.
 
-To facilitate supporting so many languages, the AWS CDK is developed in one language (TypeScript). Language bindings are generated for the other languages through the use of a tool called [JSII](https://aws.github.io/jsii/).
+To facilitate supporting so many languages, the AWS PDK is developed in one language (TypeScript). Language bindings are generated for the other languages through the use of a tool called [JSII](https://aws.github.io/jsii/).
```
Comment (mteichtahl) [inline docs/content/getting_started/index.md]: Remove "so"

File: docs/content/getting_started/index.md [hunk 2]
```diff
@@ -101,7 +101,49 @@ The AWS PDK lets you define your project structure as code in one of its support
...
+    ```bash
+    <dependency>
+        <groupId>software.aws</groupId>
+        <artifactId>pdk</artifactId>
+        <version>LATEST</version>
+    </dependency>
+    ```
+
+!!!tip
+    Whilst the AWS PDK can be used directly via these package managers, we recommend bootstrapping via the `pdk new` command which negates the need to interact directly with these package managers.
```
Comment (mteichtahl) [inline docs/content/getting_started/index.md]: May want to point to prerequisites to let the reader know they need to install at least one

**Review by mteichtahl — DISMISSED** [review]

File: docs/content/getting_started/index.md (hunk 1 above)
Comment (agdimech) [reply docs/content/getting_started/index.md]: Updated.

File: docs/content/getting_started/index.md (hunk 2 above)
Comment (agdimech) [reply docs/content/getting_started/index.md]: Done.

**Pull request merged**